stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1619,7 +1620,7 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
                     memcpy(buffer.pointee.memory, $0.baseAddress!, hashRange.count)
                 }
             }
-            hashValue = Int(bitPattern: CFHashBytes(buffer.pointee.memory.assumingMemoryBound(to: UInt8.self), hashRange.count))
+            hashValue = Int(bitPattern: CFHashBytes(buffer.pointee.memory.bindMemory(to: UInt8.self, capacity: hashRange.count), hashRange.count))
```

Comment:
Ah. Here's were I've led you astray. The caller of `hashValue` would not expect memory to be rebound to a particular type. That's purely an implementation detail. The right answer in this case is debatable, but here's what I thing. It should remain as it was, using `assumingMemoryBound` with a comment along these lines:
"assumingMemoryBound converts Data's raw pointer into the 'uint8_t*' type required by CFHashBytes. This is safe because CFHashBytes is known to be implemented in C, so the Swift compiler will never see an access to the `UInt8` pointer. `bindMemory` is avoided here because the the caller could potentially access this Data object's underlying memory using a different type (e.g. via a `bytesNoCopy` initializer).